### ISSUE TITLE
Xenial jobs failing in FastCDR

### ISSUE THREAD

**Issue by wjwwood** [ros2/build_farmer#143]
Starting last night (July 26th) many of the Xenial nightly jobs are failing:

```
--- output: fastcdr
10:30:41 Not searching for unused variables given on the command line.
10:30:41 -- The C compiler identification is GNU 5.4.0
10:30:41 -- The CXX compiler identification is GNU 5.4.0
10:30:41 -- Check for working C compiler: /usr/lib/ccache/cc
10:30:41 -- Check for working C compiler: /usr/lib/ccache/cc -- works
10:30:41 -- Detecting C compiler ABI info
10:30:41 -- Detecting C compiler ABI info - done
10:30:41 -- Detecting C compile features
10:30:41 -- Detecting C compile features - done
10:30:41 -- Check for working CXX compiler: /usr/lib/ccache/c++
10:30:41 -- Check for working CXX compiler: /usr/lib/ccache/c++ -- works
10:30:41 -- Detecting CXX compiler ABI info
10:30:41 -- Detecting CXX compiler ABI info - done
10:30:41 -- Detecting CXX compile features
10:30:41 -- Detecting CXX compile features - done
10:30:41 -- Configuring Fast CDR
10:30:41 -- Version: 1.0.7
10:30:41 -- To change de version modify the file configure.ac
10:30:41 -- Performing Test SUPPORTS_CXX14
10:30:41 -- Performing Test SUPPORTS_CXX14 - Success
10:30:41 -- Check if the system is big endian
10:30:41 -- Searching 16 bit integer
10:30:41 -- Looking for sys/types.h
10:30:41 -- Looking for sys/types.h - found
10:30:41 -- Looking for stdint.h
10:30:41 -- Looking for stdint.h - found
10:30:41 -- Looking for stddef.h
10:30:41 -- Looking for stddef.h - found
10:30:41 -- Check size of unsigned short
10:30:41 -- Check size of unsigned short - done
10:30:41 -- Using unsigned short
10:30:41 -- Check if the system is big endian - little endian
10:30:41 -- Configuring done
10:30:41 CMake Error at cmake/common/check_configuration.cmake:119 (target_compile_options):
10:30:41   Error evaluating generator expression:
10:30:41 
10:30:41     $<VERSION_GREATER_EQUAL:$<CXX_COMPILER_VERSION>,6.0.0>
10:30:41 
10:30:41   Expression did not evaluate to a known generator expression
10:30:41 Call Stack (most recent call first):
10:30:41   src/cpp/CMakeLists.txt:51 (set_common_compile_options)
10:30:41 
10:30:41 
10:30:41 -- Generating done
10:30:41 -- Build files have been written to: /home/rosbuild/ci_scripts/ws/build/fastcdr
10:30:41 ---
10:30:41 
```

-- from https://ci.ros2.org/view/nightly/job/nightly_xenial_linux-aarch64_release/55/console#console-section-22

Interestingly, the normal Linux job passed, but the others failed:

- Passed:
  - https://ci.ros2.org/view/nightly/job/nightly_xenial_linux_release/51/
- Failed:
  - https://ci.ros2.org/view/nightly/job/nightly_xenial_linux-aarch64_release/55/
  - https://ci.ros2.org/view/packaging/job/packaging_xenial_linux/45/
  - https://ci.ros2.org/view/packaging/job/packaging_xenial_linux-aarch64/45/

Maybe something was merged in between them running last night. I'll check that first.

**Comment by wjwwood** [comment]
The passing linux job is using `15f46e820780bd92eaa2d10675b744b67a9ad70e` and the failing one is `63202c16739dc527f8ba7456847bb8dcb7ddab5b`, I'll investigate and report upstream.

**Comment by wjwwood** [comment]
It was a commit last night which started using a feature only available in CMake >= 3.7: https://github.com/eProsima/Fast-CDR/issues/21

**Comment by dirk-thomas** [comment]
Fixed upstream.
